gc/gc-base/src/main/java/org/projectnessie/versioned/persist/gc/IcebergExpiredContentValues.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.persist.gc;
+
+import java.util.HashSet;
+import java.util.Set;
+import org.projectnessie.model.Content;
+import org.projectnessie.model.IcebergTable;
+
+public final class IcebergExpiredContentValues extends ExpiredContentValues {
+
+  private final Set<Long> expiredSnapshotIds = new HashSet<>();
+  private final Set<Long> liveSnapshotIds = new HashSet<>();
+
+  @Override
+  protected void addValue(Content content, boolean isExpired) {
```

Comment:
This seems like overly complex logic to me. Why not add to live if live and expired if expired then take the difference?

Additionally, I don't think this needs to be iceberg specific. Somethign is live if the commit is live and the key/contentId exists on that commit irrespective of the snapshotid. I am thinking of something closer to https://github.com/projectnessie/nessie/blob/nessie-0.9.1/versioned/tiered/gc/src/main/java/org/projectnessie/versioned/tiered/gc/IdentifyUnreferencedValues.java#L93 where we could build a bloom filter of all valid key/hash pairs then iterate over values to find out what can be deleted...then we can associate with actual concerete Content types after we find out if they can be deleted. I believe this saves a bit of database access too.

Reply:
> This seems like overly complex logic to me. Why not add to live if live and expired if expired then take the difference?

This is a small storage optimization as content will be present in any one of the set at given point of time. So, no duplicate storage.

Reply:
I have added a `BaseExpiredContentValues` now, if content type specific implementation is required, we can plug that in as well.